tools/run_examples.py

```diff
@@ -0,0 +1,190 @@
+import argparse
```

Comment:
This may be beyond the scope of this PR, but I am curious how you plan to maintain this file. For example, there are several logics in this file, are there any plans to document them or provide tests for them?

Reply:
I think if we operate examples CI with this script, we ought to update this regularly, which I think enough.
Also, I basically think this script should be maintained in a more casual way compared to optuna source code because
the number of users & readers is limited, I expect them to be familiar with Optuna's CI, this script is ostensibly Python: shell script with convenient Python's argparse and string functions.

> For example, there are several logics in this file, are there any plans to document them or provide tests for them?

By document, do you mean we mention this script in Optuna's API documentation or somewhere else? Or do you mean I should write comments more in this script?
As to tests, do you have any concrete and reasonable test? If you want to check the selected tests are correct or not, we're going to maintain the lists of examples compatible with specific Python versions. If we do this, it's better to maintain that those lists in this script.